### ISSUE TITLE
GPU Usage fails to display when hwmon isn't found where expected

### ISSUE THREAD

**Issue by nuudlman** [flightlessmango/MangoHud#757]
It seems the change to prefer hwmon over gpu_metrics [(commit)](https://github.com/flightlessmango/MangoHud/commit/b30302c889eed2d01744411233a041a0491cfba9) breaks when hwmon is only enumerated under `card0` and not `card0-<display>`

system:
* Fedora 36, Gnome 42
* Intel i7-6700k
* 16GB DDR4 @ JEDEC
* AMD Radeon RX 570 8GB (polaris10, LLVM 14.0.0, DRM 3.44, 5.17.5-301.fsync.fc36.x86_64)
* Mesa 23.0.0-git (20220508.d2ab0ed), RADV
* MangoHud 0.6.7

`/sys/class/drm`:
```
card0  card0-DP-2  card0-DP-3  card0-DP-4  card0-DVI-D-1  card0-HDMI-A-3  card1  card1-DP-1  card1-HDMI-A-1  card1-HDMI-A-2  renderD128  renderD129  version
```
plain `card0` enumerates hwmon:
`/sys/class/drm/card0/device/hwmon/hwmon0/`:
```
device       fan1_input  fan1_min     freq1_input  freq2_input  in0_input  name   power1_average  power1_cap_default  power1_cap_min  pwm1         pwm1_max  subsystem   temp1_crit_hyst  temp1_label
fan1_enable  fan1_max    fan1_target  freq1_label  freq2_label  in0_label  power  power1_cap      power1_cap_max      power1_label    pwm1_enable  pwm1_min  temp1_crit  temp1_input      uevent
```
whereas `card0-<display>` doesn't:
`/sys/class/drm/card0-DP-3`:
```
ddc  device  dpms  drm_dp_aux2  edid  enabled  i2c-10  modes  power  status  subsystem  uevent
```
leading to the MangoHud error:
```
2022-05-10 22:08:52.198] [MANGOHUD] [info] [config.cpp:112] skipping config: '/usr/bin/MangoHud.conf' [ not found ]
[2022-05-10 22:08:52.198] [MANGOHUD] [info] [config.cpp:112] skipping config: '/home/isaac/.config/MangoHud/vkcube.conf' [ not found ]
[2022-05-10 22:08:52.198] [MANGOHUD] [info] [config.cpp:117] parsing config: '/home/isaac/.config/MangoHud/MangoHud.conf'
]Selected GPU 0: AMD RADV POLARIS10, type: DiscreteGpu
[2022-05-10 22:08:52.283] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-DVI-D-1/device/hwmon/': No such file or directory
[2022-05-10 22:08:52.283] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card1-HDMI-A-1/device/hwmon/': No such file or directory
[2022-05-10 22:08:52.283] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-DP-4/device/hwmon/': No such file or directory
[2022-05-10 22:08:52.283] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-DP-2/device/hwmon/': No such file or directory
[2022-05-10 22:08:52.284] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card1-HDMI-A-2/device/hwmon/': No such file or directory
[2022-05-10 22:08:52.284] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card1-DP-1/device/hwmon/': No such file or directory
[2022-05-10 22:08:52.284] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-DP-3/device/hwmon/': No such file or directory
[2022-05-10 22:08:52.284] [MANGOHUD] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-HDMI-A-3/device/hwmon/': No such file or directory
```
and the lack of gpu metrics:
![изображение](https://user-images.githubusercontent.com/62861466/167764580-92027ce0-6d98-40b4-98df-1703a522534d.png)

**Comment by jackun** [comment]
It should be fixed in https://github.com/flightlessmango/MangoHud/commit/bd2f7b24b896e6132d48910cf94fd8ece9626bd8 (unless it's some other logic error :P)

**Comment by nuudlman** [comment]
I'll build and test it tonight, hopefully that was it.

**Comment by flightlessmango** [comment]
Are you sure you actually compiled the right versions? #756 had the same issue on a 570 polaris card and was fixed by bd2f7b24b896e6132d48910cf94fd8ece9626bd8

**Comment by nuudlman** [comment]
That was it, build.sh worked and the overlay works now
